### PR TITLE
[core] Delete redundant if statement when produce changelog by lookup for first row.

### PR DESCRIPTION
**Pull request by wwj6591812**
### Purpose

When I study the logic of produce changelog, I find a redundant if statement affects my understanding of the code logic.
In order not to affect more people, so now delete it.

Linked issue: close #xxx

### Tests

Existing tests should cover this change.

### CODE REVIEW

**Comment by wwj6591812** [comment]
The ut is fail, but not caused by my modify.

File: paimon-core/src/main/java/org/apache/paimon/mergetree/compact/FirstRowMergeFunctionWrapper.java [hunk 1]
```diff
@@ -67,10 +67,10 @@ public ChangelogResult getResult() {
         }
 
         reusedResult.setResult(result);
-        if (result.level() == 0) {
-            // new record, output changelog
-            return reusedResult.addChangelog(result);
-        }
+
+        // mergeFunction.containsHighLevel is false, so new record, output changelog
```
Comment (Aitozi) [inline paimon-core/src/main/java/org/apache/paimon/mergetree/compact/FirstRowMergeFunctionWrapper.java:71]: what about 

```
// new record, output changelog
return reusedResult.setResult(result).addChangelog(result);
```

Comment (wwj6591812) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/compact/FirstRowMergeFunctionWrapper.java]: done
good, very thanks.

**Review by Aitozi — APPROVED** [review]
LGTM

File: paimon-core/src/main/java/org/apache/paimon/mergetree/compact/FirstRowMergeFunctionWrapper.java [hunk 2]
```diff
@@ -66,11 +66,7 @@ public ChangelogResult getResult() {
             return reusedResult;
         }
 
-        reusedResult.setResult(result);
-        if (result.level() == 0) {
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/mergetree/compact/FirstRowMergeFunctionWrapper.java:70]: Why removing this `if`? What about merging level 4 and level 5?

Comment (Aitozi) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/compact/FirstRowMergeFunctionWrapper.java:70]: if merging with high level, it should return at line 61.

**Comment by JingsongLi** [comment]
+1

**Pull request merged**